jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -437,6 +437,141 @@ def vue_do_aper_phot(self, *args, **kwargs):
             self.result_available = True
             self.plot_available = True
 
+    def unpack_batch_options(self, **options):
+        """
+        Unpacks a dictionary of options for batch mode, including all combinations of any values
+        passed as tuples or lists.  For example::
+
+            unpack_batch_options(dataset=['image1', 'image2'],
+                                 subset=['Subset 1', 'Subset 2'],
+                                 bg_subset=['Subset 3'],

```

Comment:
So, if `bg_subset` has multiple elements, then it is unpacked too? Is that what people want though? Usually a given background is very specific to the chosen aperture?

Reply:
Also, this bypasses the UI checks, right? For example, now someone can define "Subset 1" to be both the aperture and background and we cannot stop them.

Reply:
> So, if `bg_subset` has multiple elements, then it is unpacked too? Is that what people want though? Usually a given background is very specific to the chosen aperture?

Right.  If you want specific combinations, then you call the other method (or call this first, and pass just the combinations you want).

> Also, this bypasses the UI checks, right?

If there are checks that are only in the UI, then yes, those are bypassed.  We should move (or at least copy) those to the python end.  This PR doesn't yet validate the input _values_, so technically a batch could get halfway through before erroring out, in which case you'll get the results that completed before the error.  We'll need to think of what we want to happen in a case like this.  Full validation in advance would likely require a significant refactor of the plugin code and since computing isn't too expensive, I think this is probably okay for now.

Reply:
p.s. I thought maybe we can pass multiple apertures straight into photutils instead of looping through them, but looks like that is only possible if the aperture shape/size is identical and only differ in center positions, so I guess that is not very useful for us... See https://github.com/astropy/photutils/issues/1615

> so technically a batch could get halfway through before erroring out, in which case you'll get the results that completed before the error

That would be annoying. I prefer returning all successful results and a list of combos that failed, after the loop completes.

Reply:
That would work for me and is easier to implement than full input-validation before running anything.  I'm always in favor or raising exceptions, but I can at least wait until all that can pass do so, and then raise the exception at the end so the user is aware that not all were successful.